modules/joex/src/main/scala/docspell/joex/scanmailbox/ScanMailboxTask.scala

```diff
@@ -167,6 +176,15 @@ object ScanMailboxTask {
           .map(_.toRight(new Exception(s"Folder '$name' not found")))
           .mapF(_.rethrow)
 
+    def searchMailsRecursively[C](
+        a: Access[F, C]
+    )(folder: MailFolder): MailOp[F, C, SearchResult[MailHeader]] =
+      for {
+        subFolders <- a.listFoldersRecursive(Some(folder))
+        foldersToSearch = Vector(folder) ++ subFolders
+        search <- foldersToSearch.traverse(searchMails(a))
+      } yield searchResultMonoid.combineAll(search)
```

Comment:
This is totally fine as it is. just fyi: you can have the compiler to "look up" a monoid instance for some type (because how it is implemented in cats) with this: `Monoid[SearchResult]` that would do just the same, scala looks by default in the companion object for stuff. As an alternative, cats provides some syntax enhancements that you get by importing `cats.implicits._` (this gets all, you can also import à la carte). With this import you should be able to do `search.foldMonoid`.